[Selectable Components: Structured List - Usage] Website Doc (#3967)
carbon-design-system/carbon-website: src/pages/components/structured-list/usage.mdx

<Row>
<Column colLg={12}>

> ![structured list anatomy](images/structured-list-usage-1.png)

Author: I've made the picture as 522 px.

| _Hover_    | When a user’s mouse cursor is hovering over the structured list item.                                                                                                                                                             |
| _Focus_    | When a user tabs or clicks on a row, it becomes focused, indicating the user has successfully navigated to the component (except default structured list).                                                                        |
| _Selected_ | When a user presses space or clicks on a structured list row to select it.                                                                                                                                                        |
| _Disabled_ | When the user is not allowed to interact with a selectable structured list due to either permissions, dependencies, or pre-requisites.                                                                                            |

> | _Skeleton_ | When the user can review but not modify the component. This state removes all interactive functions like the disabled state but can still be focusable, accessible by screen readers, and passes visual contrast for readability. |

Reviewer: This seems to still need to be changed @Kritvi-bhatia17

> | _Skeleton_ | When the user can review but not modify the component. This state removes all interactive functions like the disabled state but can still be focusable, accessible by screen readers, and passes visual contrast for readability. |

Author: Ohh got it!

## Interactions

#### Selectable structured list

- Only one item can be selected from the list.
- By default, no option should be selected.
- If you need to select multiple items, use a
  [data table.](/components/data-table/usage)
- When the user selects an item from the list, the selected row will appear with
  the `radio-button--checked` icon.

## Default

Default structured lists are used to display information to the user, such as

> pricing plans. These structured list can have internal calls-to-action (CTAs),

Reviewer: This still needs to be updated

> pricing plans. These structured list can have internal calls-to-action (CTAs),

Author: Oh shit sorry I guess I missed it, done!

such as a [button](/components/button/usage) or
a [link](/components/link/usage).

<Row>
<Column colLg={12}>

> ![default structured list](images/structured-list-usage-4.png)

Reviewer: Just some typos:

- In the left most column, the word "Related" has a typo.
- There is an extra space in the first paragraph between "Director" and "platform".
- There is an extra space in the first paragraph between "hosting," and "operations,".
- There is an extra space in the first paragraph between "can" and "quickly".
- Under the "Single-tenant" column in the structured list, the first sentence should say the word "require" not the word required.
- Under the "Multi-tenant" column in the structured list,  in the second row, the beginning of the sentence needs to be capitalized and be "On-demand".

<Caption> Default structured list </Caption>

</Column>
</Row>

## Selectable

Selectable structured lists work well for presenting options to a user in a
structured manner, such as a set of pricing plans. Selectable tiles may contain
internal CTAs (like links to docs) if the internal CTA is given its own click
target. Selectable structured lists have a single-select state working like
a [radio button](/components/radio-button/usage).

<Row>
<Column colLg={12}>

> ![selectable structured list](images/structured-list-usage-5.png)

Reviewer: I made some comments in your Figma file about some spacing in one of the rows in the structured list in this image.

There are also a few typos: 
- In the first left column there is a typo in the word "Category".
- In the right panel, there is a typo in the word "Washington".
- In the right panel, in the terms, the word should be "agreed" and not agree.

The radio icons are also not aligned correctly.

Reviewer: There is still a typo in the word "Washington" in the right panel.

Author: Ohh sorry, yes done!

<Caption> Selectable structured list </Caption>

</Column>
</Row>

## Modifiers

### Background

> Background has been added to the structured list, and it is optional. The

Reviewer: ```suggestion
A background has been added to the structured list, and it is optional. The
```

> background is only present for the hang alignment structured list, both default

Reviewer: ```suggestion
background is only available for the hang alignment structured list in default
```

> and selectable; it is not included for flush alignment.

Reviewer: ```suggestion
and selectable variants. It is not available for the flush alignment structured list.
```
